ReactiveCocoa/NSObject+KeyValueObserving.swift

```diff
@@ -55,6 +55,105 @@ extension Reactive where Base: NSObject {
 			return disposable
 		}
 	}
+
+	#if swift(>=3.2)
+	private func producer<U>(
+		for keyPath: KeyPath<Base, U>,
+		transform: @escaping (Any?) -> U
+	) -> SignalProducer<U, NoError> {
+		return SignalProducer { observer, lifetime in
+			let disposable = KeyValueObserver.observe(
+				self.base,
+				keyPath: keyPath._kvcKeyPathString!,
```

Comment:
I'm not comfortable with the use of `_kvcKeyPathString` here.

Besides the fact that it's underscore-prefixed to indicate that it's somewhat private, this is `nil` for some some `KeyPath`s—even those on `NSObject`s. The property itself needs to be `dynamic` in order for this to be non-`nil`.

We need to use the built-in `KeyPath` KVO API.

Reply:
We can wrap the official closure-based KVO API, but it does not support observing weak properties (and properties of these) as the RAC version does.

Reply:
Either way I don't have a strong opinion, since I haven't been using KVO for a while. Even when I am using it, I don't recall any need of observing weak properties.

Reply:
FWIW, I have some observations on weak properties all over the place.

Wouldn't it make sense to add a variant that uses `ReferenceWritableKeyPath` for those that want to support weak references? That'll allow us to use the "official" API.

Reply:
The actual problem is that the official one has not supported emitting `nil` when weak references are zeroed. I am not aware that the mechanics have now accommodated this, and the "new" KVO API is just a wrapper around what we have used for ages.

Reply:
> The actual problem is that the official one has not supported emitting nil when weak references are zeroed.

@andersio I just verified that this is the case in a playground. Yuck!

Reply:
> Besides the fact that it's underscore-prefixed to indicate that it's somewhat private, this is nil for some some KeyPaths—even those on NSObjects. The property itself needs to be dynamic in order for this to be non-nil.

@mdiep I was not aware of that until I tried it yesterday, but the same is already required even in RAC with the current String based `DynamicProperty`.

Apparently KVO in Swift as a whole only works for properties that are `@dynamic`, no matter if you use String based `KeyPath`s or the new smart `KeyPaths`.

```
class NSPerson: NSObject {
    @objc dynamic var firstName: String = ""
    @objc var lastName: String = ""
}
let person = NSPerson()


person.observe(\NSPerson.firstName) { _, _ in print("KVO: \(person.firstName)")}
person.observe(\NSPerson.lastName) { _, _ in print("KVO: \(person.lastName)")}

let firstName = DynamicProperty<String>(object: person, keyPath: "firstName")
let lastName = DynamicProperty<String>(object: person, keyPath: "lastName")

firstName.signal.observeValues { print("DynamicProperty: \($0)")}
lastName.signal.observeValues { print("DynamicProperty: \($0)")}

person.firstName = "John"
person.lastName = "Doe"
```

This prints the following:

```
DynamicProperty: John
KVO: John
```

So neither plain Swift 4 Smart Key Path KVO nor the existing String based RAC DynamicProperty work for properties  that are missing the `dynamic` even if they are `@objc`